### PR TITLE
Add integration tests for compression

### PR DESCRIPTION
**Pull request by ofalvai**
### Checklist

- [ ] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [ ] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a no [version update](https://semver.org/)

### Context

Covering the archiving and compression code in the step with tests. Unit tests would only test how we call external tools, while E2E testing this logic would be cumbersome.

### Changes

Add `integration` folder with integration tests tagged as `integration`. These tests don't run with the simple `go test` command, only when the tag is specified.

Extract compression logic into separate package so that it's testable independently of the rest of the step.

### CODE REVIEW

File: integration/compression_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-utils/v2/env"
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-steplib/steps-save-cache/compression"
+)
+
+func Test_compression(t *testing.T) {
```
Comment (hisaac) [inline integration/compression_test.go:19]: Maybe we should add a check that the `ztsd` option is available to this, with a helpful warning if it's not.

File: step/step.go [hunk 2]
```diff
@@ -136,24 +138,8 @@ func (step SaveCacheStep) compress(paths []string) (string, error) {
...
-		"--directory",
-		step.envRepo.Get("BITRISE_SOURCE_DIR"),
-	}
-	tarArgs = append(tarArgs, paths...)
-
-	cmd := step.commandFactory.Create("tar", tarArgs, nil)
-
-	step.logger.Debugf("$ %s", cmd.PrintableCommandArgs())
-
-	out, err := cmd.RunAndReturnTrimmedCombinedOutput()
```
Comment (hisaac) [inline step/step.go:154]: Good call moving this to its own file. I'll do the same for the decompression code. 👍

**Review by hisaac — CHANGES_REQUESTED** [review]
If it's not too difficult, I think it'd be good if we add a check for `zstd` to the integration test with a helpful error message if it's not there. The error message you actually get isn't very helpful, so a developer might not know how to fix the problem.

File: integration/compression_test.go (hunk 1 above)
Comment (ofalvai) [reply integration/compression_test.go:20]: Good idea, done!

**Review by hisaac — APPROVED** [review]
Great work on this!

**Pull request merged**